osquery/tables/system/windows/windows_crashes.cpp

```diff
@@ -0,0 +1,764 @@
+  if (symbols != nullptr)
+    symbols->Release();
+  if (control != nullptr)
+    control->Release();
+  if (client != nullptr) {
+    client->SetOutputCallbacks(NULL);
+    client->EndSession(DEBUG_END_PASSIVE);
+    client->Release();
+  }
+  return;
+}
+
+/*
+Note: appears to only detect unmanaged stack frames.
+See http://blog.steveniemitz.com/building-a-mixed-mode-stack-walker-part-2/
+*/
+void logStackTrace(const char* fileName, Row& r) {
+  IDebugClient4* client;
+  IDebugControl4* control;
+  IDebugSymbols3* symbols;
+  DEBUG_STACK_FRAME stackFrames[kNumStackFramesToLog] = {0};
+  unsigned long numFrames = 0;
+  char context[1024] = {0};
+  unsigned long type = 0;
+  unsigned long procID = 0;
+  unsigned long threadID = 0;
+  unsigned long contextSize = 0;
+
+  // Create interfaces
+  if (DebugCreate(__uuidof(IDebugClient), (void**)&client) != S_OK) {
+    LOG(ERROR) << "DebugCreate failed while debugging crash dump: " << fileName;
+    return debugEngineCleanup(client, nullptr, nullptr);
+  }
+  if ((client->QueryInterface(__uuidof(IDebugControl4), (void**)&control) !=
+       S_OK) ||
+      (client->QueryInterface(__uuidof(IDebugSymbols), (void**)&symbols) !=
+       S_OK)) {
+    LOG(ERROR) << "QueryInterface failed while debugging crash dump: "
+               << fileName;
+    return debugEngineCleanup(client, control, symbols);
+  }
+
+  // Initialization
+  if (symbols->SetImagePath(r["path"].c_str()) != S_OK) {
+    LOG(ERROR) << "Failed to set image path to \"" << r["path"]
+               << "\" while debugging crash dump: " << fileName;
+    return debugEngineCleanup(client, control, symbols);
+  }
+  if (symbols->SetSymbolPath("srv*C:\\Windows\\symbols*http://"
+                             "msdl.microsoft.com/download/symbols") != S_OK) {
+    LOG(ERROR) << "Failed to set symbol path while debugging crash dump: "
+               << fileName;
+    return debugEngineCleanup(client, control, symbols);
+  }
+  if (client->OpenDumpFile(fileName) != S_OK) {
+    LOG(ERROR) << "Failed to open dump file while debugging crash dump: "
+               << fileName;
+    return debugEngineCleanup(client, control, symbols);
+  }
+  if (control->WaitForEvent(DEBUG_WAIT_DEFAULT, INFINITE) != S_OK) {
+    LOG(ERROR) << "Initial processing failed while debugging crash dump: "
+               << fileName;
+    return debugEngineCleanup(client, control, symbols);
+  }
+
+  // Get stack frames from dump
+  if (control->GetStoredEventInformation(&type,
+                                         &procID,
+                                         &threadID,
+                                         context,
+                                         sizeof(context),
+                                         &contextSize,
+                                         NULL,
+                                         0,
+                                         0) == S_OK) {
+    auto contextData = new char[kNumStackFramesToLog * contextSize];
```

Comment:
While your use of resources here looks correct, any chance this could be wrapped in a `unique_ptr` or even a `std::vector<char>`? We try to avoid direct memory manipulation wherever possible. [Here is an example with `std::vector`](https://github.com/facebook/osquery/blob/master/osquery/tables/system/windows/processes.cpp#L189)